pkg/instancetype/instancetype.go

```diff
@@ -932,6 +934,29 @@ func validateCPU(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.Vir
 	return conflicts
 }
 
+func applyNodeSelector(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
+	if vmiSpec.NodeSelector != nil {
+		return Conflicts{field.Child("nodeSelector")}
+	}
+
+	vmiSpec.NodeSelector = make(map[string]string, len(instancetypeSpec.NodeSelector))
+	for k, v := range instancetypeSpec.NodeSelector {
+		vmiSpec.NodeSelector[k] = v
+	}
+
+	return nil
+}
+
+func applySchedulerName(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) Conflicts {
+	if vmiSpec.SchedulerName != "" {
```

Comment:
`instancetypeSpec.SchedulerName != nil && vmiSpec.SchedulerName != ""`

^ assuming `instancetypeSpec.SchedulerName` is moved to a string pointer below.